setup.py

```diff
@@ -3,12 +3,25 @@
 
 import sys
 
+from distutils.version import LooseVersion
+
+# We require setuptools 30.3.0 or later for the configuration in setup.cfg to
+# work properly.
+import setuptools
+if LooseVersion(setuptools.__version__) < LooseVersion('30.3.0'):
+    sys.stderr.write("ERROR: Astropy requires setuptools 30.3.0 or later "
+                     "(found {0})".format(setuptools.__version__))
+    sys.exit(1)
+
+from setuptools.config import read_configuration
+conf = read_configuration('setup.cfg')
+
 # This is the same check as astropy/__init__.py but this one has to
 # happen before importing ah_bootstrap
-__minimum_python_version__ = '3.5'
-if sys.version_info < tuple((int(val) for val in __minimum_python_version__.split('.'))):
+minimum_python_version = conf['options']['python_requires'].replace('>', '').replace('<', '').replace('=', '')
```

Comment:
Why replace the `<` - that should error, no?

Reply:
Triple replace is messing with my head. What is this trying to do?

Reply:
Basically this is changing a requirement (from e.g. ``>=3.5``) to an actual minimum version (``3.5``). I agree ``<`` shouldn't really be used though.